### PR TITLE
change: Make VQE_Chemistry use parameterized circuits.

### PR DESCRIPTION
**Pull request by math411**
*Issue #, if available:*

*Description of changes:*
Make VQE_Chemistry use parameterized circuits.

Tested in a notebook pulling in the branch from the sdk:

```
!pip install git+https://github.com/aws/amazon-braket-sdk-python.git@param_circuits
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by speller26 — COMMENTED** [review]
[Invalid notebook](https://github.com/aws/amazon-braket-examples/blob/param_circ_ex/examples/hybrid_quantum_algorithms/VQE_Chemistry/VQE_chemistry_braket.ipynb); might be a JSON formatting problem.

File: examples/hybrid_quantum_algorithms/VQE_Chemistry/VQE_chemistry_braket.ipynb [hunk 1]
```diff
@@ -517,13 +517,12 @@
...
     "# Construct circuit for UCCSD ansatz parameterized by VQE parameter per McArdle et al. \n",
-    "def uccsd_ansatz(a_theta):\n",
+    "def uccsd_ansatz():\n",
     "    \"\"\" Construct UCCSD ansatz circuit parameterized by a_theta.\n",
-    "    Parameters:\n",
-    "        a_theta [float]: VQE parameter to be optimized\n",
     "    Returns:\n",
     "        ansatz_ckt [braket.Circuit]: Ansatz circuit\n",
     "    \"\"\"\n",
+    "    a_theta = FreeParameter("a_theta")\n",
```
Comment (speller26) [inline examples/hybrid_quantum_algorithms/VQE_Chemistry/VQE_chemistry_braket.ipynb:525]: Looks like this extra quote might be the problem.

File: examples/hybrid_quantum_algorithms/VQE_Chemistry/VQE_chemistry_braket.ipynb [hunk 2]
```diff
@@ -352,7 +352,7 @@
     "from openfermion import MolecularData\n",
```
Comment (speller26) [inline examples/hybrid_quantum_algorithms/VQE_Chemistry/VQE_chemistry_braket.ipynb:352]: <pre class="ql-syntax" spellcheck="false">Line #26.    ansatz_ckt_zero = ansatz_ckt(0)
</pre><p>Why not just print the circuit with the unbound free parameter? Then follow up with a little blurb about binding the parameter, and <em>then</em> run the circuit. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-braket-examples/pull/114/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (speller26) [inline examples/hybrid_quantum_algorithms/VQE_Chemistry/VQE_chemistry_braket.ipynb:352]: <pre class="ql-syntax" spellcheck="false">Line #2.    ansatz_ckt = uccsd_ansatz(t)
</pre><p>Why is this needed? What is <code>t</code>?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-braket-examples/pull/114/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (math411) [reply examples/hybrid_quantum_algorithms/VQE_Chemistry/VQE_chemistry_braket.ipynb:352]: Removed

Comment (math411) [reply examples/hybrid_quantum_algorithms/VQE_Chemistry/VQE_chemistry_braket.ipynb:352]: Fixed

Comment (speller26) [inline examples/hybrid_quantum_algorithms/VQE_Chemistry/VQE_chemistry_braket.ipynb:352]: <pre class="ql-syntax" spellcheck="false">Line #2.    def uccsd_ansatz():
</pre><p>Now that the circuit is inherently parametrized, I don't think it needs to be defined in a method; this can all be moved inline:</p><pre class="ql-syntax" spellcheck="false">a_theta = FreeParameter("a_theta")
# Initialize HF state |0011&gt;
ansatz_ckt = Circuit().x(2).x(3)
# Perform intial rotations to measure in Y &amp; X bases
ansatz_ckt.rx(3,np.pi/2.).h(range(3))
# Entangle with CNOTs
ansatz_ckt.cnot(3,2).cnot(2,1).cnot(1,0)
# Perform the rotation in Z-basis
ansatz_ckt.rz(0,a_theta)
# Uncompute the rotations
ansatz_ckt.cnot(1,0).cnot(2,1).cnot(3,2)
ansatz_ckt.h(range(3))
ansatz_ckt.rx(3,-np.pi/2.)
</pre><p>All subsequent uses of <code>ansatz_ckt</code> can now just use this circuit directly, rather than require a method call. As well, might be worth just renaming the <code>ansatz_ckt</code> object to the more descriptive <code>ansatz_uccsd</code>.    </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-braket-examples/pull/114/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (math411) [reply examples/hybrid_quantum_algorithms/VQE_Chemistry/VQE_chemistry_braket.ipynb:352]: Good callout - made these changes.

**Review by speller26 — APPROVED** [review]
